### PR TITLE
Feature/benchmark global optimization algorithm

### PR DESCRIPTION
**Pull request by ronaldpereira**
# Pull Request

## Description

This PR integrates a new **Global Optimization Algorithm** for benchmark generation that overcomes the limitations of the existing greedy optimization approach. The global optimizer uses multiple advanced strategies to avoid local maxima and find truly optimal builds.

### Added
* **Global Optimization Algorithm** with 4 strategies:
  - Genetic Algorithm with crossover and mutation
  - Simulated Annealing with adaptive temperature cooling
  - Multi-start Greedy optimization from different starting points
  - Hybrid optimization combining multiple approaches
* **UI Selection Interface**: Radio button in benchmark settings to choose between "Greedy" and "Global" optimization
* **Worker Integration**: Modified `computeOptimalSimulationWorker.ts` to conditionally use global or greedy optimization
* **Orchestrator Support**: Updated `BenchmarkSimulationOrchestrator` class with optimization algorithm selection
* **Complete Internationalization**: Added "OptimizationAlgorithm" translations for all 13 supported locales
* **Type Safety**: Extended interfaces (`BenchmarkForm`, `ComputeOptimalSimulationWorkerInput`) with `optimizationAlgorithm` field

### Changed
* **Benchmark Workflow**: Modified data flow from UI form → orchestrator → worker to pass algorithm choice throughout the pipeline
* **Default Behavior**: Set "Greedy" as default algorithm to maintain backward compatibility
* **Test Utilities**: Updated benchmark test utilities to include algorithm selection

### Technical Details
The global optimizer addresses key limitations of the greedy approach:
- **Local Maxima Avoidance**: Uses population-based and probabilistic methods
- **Better Exploration**: Genetic algorithm explores solution space more thoroughly
- **Adaptive Strategy**: Simulated annealing provides controlled randomness with cooling
- **Robustness**: Multi-start approach reduces dependency on initial conditions
- **Performance Balance**: Hybrid method combines best aspects of all strategies

### Performance Impact
- Global optimization takes longer but finds significantly better solutions
- Users can choose based on their preference for speed vs. optimality
- Greedy algorithm remains default for existing workflows

## Related Issue

* Addresses the fundamental limitation of greedy optimization getting stuck in local maxima
* Enhances benchmark generation quality for more accurate character build evaluation

## Checklist

- [x] I have added commit messages that are descriptive and meaningful.
- [x] I have tested the changes locally.
- [x] I have reviewed the code changes.

### Benchmark Settings UI
The new optimization algorithm selection appears as radio buttons in the benchmark settings panel:

**Before**: Only greedy optimization was available (implicit)
**After**: Users can choose between "Greedy" (fast) and "Global" (optimal) algorithms

### Algorithm Selection Interface
- **Greedy**: Fast optimization using incremental improvements (default)
- **Global**: Advanced optimization using genetic algorithm, simulated annealing, and multi-start strategies

### Internationalization Support
The "Optimization Algorithm" label is properly translated across all supported locales:
- English: "Optimization algorithm" 
- Chinese (Simplified): "优化算法"
- Japanese: "最適化アルゴリズム"
- German: "Optimierungsalgorithmus"
- French: "Algorithme d'optimisation"
- Spanish: "Algoritmo de optimización"
- Italian: "Algoritmo di ottimizzazione"
- Korean: "최적화 알고리즘"
- Portuguese (Brazil): "Algoritmo de otimização"
- Russian: "Алгоритм оптимизации"
- Vietnamese: "Thuật toán tối ưu hóa"
- Chinese (Traditional): "最佳化演算法"
- Afar: "crwdns45370:0crwdne45370:0" (Crowdin template)

### CODE REVIEW

**Review by FI00ds — COMMENTED** [review]
For the locale files the workflow is to only ever manually modify the `en_US` source files, all other locales are managed entirely through Crowdin

**Comment by ronaldpereira** [comment]
@FI00ds Thanks for the review. I've addressed all the issues. Let me know if you spot anything else.

**Comment by ronaldpereira** [comment]
# Pull Request


## Description


**Fixed**: Local maxima issue in DPS scoring benchmarks by implementing deterministic global optimization

**Changed**: 
* Replaced 4 probabilistic optimization strategies (Genetic Algorithm, Simulated Annealing, Multi-start Greedy, Hybrid) with a single deterministic branch-and-bound algorithm
* Removed randomness from benchmark optimization to ensure reproducible and globally optimal results
* Updated `GlobalBenchmarkOptimizer` class to use systematic search space exploration instead of heuristic approaches

**Added**:
* Deterministic global optimization algorithm using constraint propagation and intelligent pruning
* Branch-and-bound optimization with mathematical upper bounds for efficient search space pruning
* Search space calculation and bounds estimation for optimization planning
* Complete solution construction from partial assignments
* Safety limits to prevent excessive computation while maintaining optimality guarantees

**Technical Details**:
* The new algorithm specifically addresses stat conversion mechanics (e.g., "for every x ATK, increase y by z%") that caused the previous probabilistic methods to get stuck in local maxima
* Uses systematic enumeration of feasible solution regions rather than random sampling
* Guarantees finding the globally optimal substat distribution without making assumptions about stat weights or diminishing returns
* Maintains performance through intelligent pruning that eliminates impossible solutions early

## Related Issue
Closes #1034 

* Fixes local maxima problem in benchmark generation where characters with stat conversion mechanics would not find optimal substat distributions
* Addresses inconsistent benchmark results due to probabilistic nature of previous optimization strategies

## Checklist

- [x] I have added commit messages that are descriptive and meaningful.
- [x] I have tested the changes locally.
- [x] I have reviewed the code changes.

---

### Performance Impact
* Reduces evaluation count through intelligent pruning while guaranteeing optimal results
* Eliminates the need to run multiple optimization strategies and compare results
* Provides deterministic, reproducible benchmark scores

### Breaking Changes
* None - the public API remains unchanged, only the internal optimization algorithm has been replaced

### Testing
* Verified algorithm correctness through unit tests of core components (branch-and-bound logic, constraint propagation, upper bound calculation)
* Confirmed proper handling of edge cases and constraint validation
* Tested performance scaling with realistic problem sizes

File: src/lib/simulations/orchestrator/benchmarkSimulationOrchestrator.ts [hunk 1]
```diff
@@ -302,7 +307,7 @@ export class BenchmarkSimulationOrchestrator {
       const mainsCount = partialSimulationWrapper.simulation.request.simFeet == Stats.SPD ? 1 : 0
       const rolls = TsUtils.precisionRound(invertDiminishingReturnsSpdFormula(mainsCount, targetSpd - finalSpeed, clonedBenchmarkScoringParams.speedRollValue), 3)
 
-      partialSimulationWrapper.speedRollsDeduction = Math.min(Math.max(0, rolls), spdRollsCap(partialSimulationWrapper.simulation, clonedBenchmarkScoringParams))
+      partialSimulationWrapper.speedRollsDeduction = Math.min(Math.max(0, Math.ceil(rolls)), spdRollsCap(partialSimulationWrapper.simulation, clonedBenchmarkScoringParams))
```
Comment (ronaldpereira) [inline src/lib/simulations/orchestrator/benchmarkSimulationOrchestrator.ts:310]: Had to add this math.ceil part as the global optimization was taking 14.95 SPD rolls as 14 and the sum of substats rolls was higher than 54 (54.95 for example). I don't think this should be allowed for neither greedy or global algorithms, but let me know if any unseen problem arises.
